src/tests/test_ncbi_taxonomy.py

```diff
@@ -0,0 +1,556 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of :mod:`ncbi_taxonomy` module.
+
+Typical usage example:
+
+    $ pytest test_ncbi_taxonomy.py
+
+"""
+
+from typing import ContextManager, Dict, List, Generator
+
+import pytest
+from pytest import param, raises
+from _pytest.fixtures import FixtureRequest
+from sqlalchemy.engine.url import make_url
+from sqlalchemy.exc import (
+    IntegrityError,
+    OperationalError,
+    SQLAlchemyError,
+    NoResultFound,
+)
+from sqlalchemy.ext.automap import automap_base
+from typing import ContextManager
+
+from ensembl.database import DBConnection, UnitTestDB
+from ensembl.ncbi_taxonomy.models import NCBITaxaName, NCBITaxaNode, NCBITaxonomy
+from ensembl.ncbi_taxonomy.api.utils import Taxonomy
```

Comment:
Pycharm imports, some were marked as not used in the file:
```suggestion
from typing import ContextManager

import pytest
from pytest import raises
from sqlalchemy.exc import (
    NoResultFound,
)

from ensembl.database import UnitTestDB
from ensembl.ncbi_taxonomy.api.utils import Taxonomy
from ensembl.ncbi_taxonomy.models import NCBITaxonomy
```